apps/services/user-profile/src/app/user-profile/userToken.controller.ts

```diff
@@ -29,4 +36,26 @@ export class UserTokenController {
   ): Promise<UserDeviceTokensDto[]> {
     return await this.userProfileService.getDeviceTokens(nationalId)
   }
+
+  @ApiOperation({
+    summary: 'admin access - returns user profile settings',
+  })
+  @Scopes(UserProfileScope.admin)
+  @ApiSecurity('oauth2', [UserProfileScope.admin])
+  @Get('userProfile/:nationalId/notification-settings')
+  @ApiOkResponse({ type: UserProfile })
+  async findOneByNationalId(
+    @Param('nationalId')
+    nationalId: string,
+  ): Promise<UserProfile> {
+    const userProfile = await this.userProfileService.findByNationalId(
```

Comment:
Could this `throw` and should be `try catched`?

Reply:
good call